File: C# Crypto Toolkit/CryptoToolkit/SecureDataTransmission.cs
using System;
using System.Numerics;

namespace AES
{
    class SecureDataTransmission
    {
        private static EllipticCurveCrypto eccMod;
        private static AES aesMod;

        private static BigInteger[] DoKeyExchange()
        {
            ECCDomain domain = ECCDomain_secp256k1.getInstance();
            eccMod = EllipticCurveCrypto.init(domain, false);
            return eccMod.DoVerifyKeyExchange();
        }

        private static void InitAES(BigInteger key, BigInteger IV)
        {
            aesMod = AES.init(key, IV);
        }

        private static string EncryptData(string plainText)
        {
            return aesMod.Encrypt(plainText);
        }

        private static string DecryptData(string cipherText)
        {
            return aesMod.Decrypt(cipherText);
        }

        private static void Main()
        {
            BigInteger[] sharedSecret = DoKeyExchange();

            InitAES(sharedSecret[0], sharedSecret[1]);

            string plainText = "It's rainy today. Please be aware!";
            Console.WriteLine("Plain text: {0}", plainText);

            string cipherText = EncryptData(plainText);
            Console.WriteLine("Cipher text in base64: {0}", cipherText);

            //Console.Write("Please input the cipher text in Base64 format: ");
            //string cipherText = Console.ReadLine();

            string recoveredText = DecryptData(cipherText);
            Console.WriteLine("Recovered plain text: {0}", recoveredText);

            Console.ReadLine();
        }
    }
}

File: Java Crypto Toolkit/SecureDataTransmission.java
package EncryptionToolkit;

import EncryptionToolkit.AES.AES;
import EncryptionToolkit.EllipticCurveCryptography.ECCDomain;
import EncryptionToolkit.EllipticCurveCryptography.ECCDomain_secp256k1;
import EncryptionToolkit.EllipticCurveCryptography.EllipticCurveCrypto;

import java.math.BigInteger;
import java.util.Scanner;

/**
 * SecureDataTransmission
 *
 * @author: Marco
 * Date: 2019/7/1 17:29
 */
public class SecureDataTransmission {
    private static EllipticCurveCrypto eccMod;
    private static AES aesMod;
    private static boolean hasInitAES = false;

    private static BigInteger[] DoKeyExchange()
    {
        ECCDomain domain = ECCDomain_secp256k1.getInstance();
        eccMod = EllipticCurveCrypto.init(domain);
        return eccMod.doVerifyKeyExchange();
    }

    private static void InitAES(BigInteger key, BigInteger IV)
    {
        aesMod = AES.initWithAESParams(key.toString(), IV.toString());
        hasInitAES = true;
    }

    private static String EncryptData(String plainText)
    {
        if (!hasInitAES)
        {
            return "";
        }
        return aesMod.encrypt(plainText);
    }

    private static String DecryptData(String cipherText)
    {
        if (!hasInitAES)
        {
            return "";
        }
        return aesMod.decrypt(cipherText);
    }

    public static void main(String[] args)
    {
        BigInteger[] sharedSecret = DoKeyExchange();

        InitAES(sharedSecret[0], sharedSecret[1]);

//        String plainText = "It's rainy today. Please be aware!";
//        System.out.println("Plain text: " + plainText);
//
//        String cipher = EncryptData(plainText);
//        System.out.println("Cipher text: " + cipher);
//
//        String recoveredText = DecryptData(cipher);
//        System.out.println("Recovered plain text: " + recoveredText);

        Scanner scanner = new Scanner(System.in);
        System.out.print("Please input the cipher text from the client: ");
        String cipherText = scanner.nextLine();

        String recoveredText = DecryptData(cipherText);
        System.out.println("Recovered plain text: " + recoveredText);
    }
}
